### DISCUSSION TITLE
Use md-editor in offline server without cdn

### DISCUSSION THREAD

**Discussion by 20** [imzbf/md-editor-v3#294] [General]
Dear @imzbf and @all!,

thank you for this wonderful software!

How can I install the editor and all dependencies without using external link. We are using in local server in rural communities that don't have internet connection.

Thank you!

**Comment by imzbf** [comment]
There is an example: [Import All Library](https://imzbf.github.io/md-editor-v3/en-US/demo#%F0%9F%99%8D%F0%9F%8F%BB%E2%80%8D%E2%99%82%EF%B8%8F%20Import%20All%20Library)

**Comment by imzbf** [comment]
Perhaps you need to save icon to your project, the solution is here: [noIconfont](https://imzbf.github.io/md-editor-v3/en-US/docs#%F0%9F%A4%9E%F0%9F%8F%BC%20noIconfont)

**Comment by OrsolDev** [comment]
compiling with: npm run build produces a lib folder, umd subfolder, index.js file.....how to use this file in my html pages in production? when I try I get an error in the console that says: argument: g, was not provided. which is logical since the main function in the index.js file waits for the arguments $e and g while it is an automatic function when opening my html page. Help me how to solve this and use the umd/index.js file in production in my website. Thanks

**Reply by imzbf** [reply]
https://imzbf.github.io/md-editor-v3/en-US/demo#%F0%9F%A4%93%20CDN

Use in html directly.

**Reply by OrsolDev** [reply]
Thank you so much
